src/org/labkey/test/util/DeferredErrorCollector.java

```diff
@@ -221,6 +210,20 @@ protected boolean isErrorDeferrable(Throwable err)
         return false;
     }
 
+    /**
+     * Record an error if the provided assertion doesn't succeed within the specified timeout.
+     *
+     * @param timeout maximum duration to wait for assertion
+     * @param wrappedAssertion {@link Runnable} that might throw an error that shouldn't fail a test immediately
+     * @return <code>true</code> if the assertion succeeded within the specified timeout
+     */
+    public boolean awaiting(Duration timeout, Runnable wrappedAssertion)
```

Comment:
This is a sensible pattern, thanks for doing this.
Existing usages of this method will need refactoring
LMK if you want me to take a crack at those, mostly they should be fresh in my mind

Reply:
Other uses are already refactored. PRs are assigned to you.